### PR TITLE
tensorflow 0.7.1 (new formula)

### PR DESCRIPTION
**Pull request by itsmeolivia**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

Added a new formula.

### CODE REVIEW

**Comment by apjanke** [comment]
Build failed due to GitHub outage. Rerunning.

File: Formula/tensorflow.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class Tensorflow < Formula
+  desc "Machine Intelligence Library"
+  homepage "https://tensorflow.org"
+  url "https://github.com/tensorflow/tensorflow.git",
+    :tag => "v0.7.1",
+    :revision => "028d0b46004c921acd48fdd0ec18128d79e18bf4",
+    :shallow => false
+
+  env :std
```
Comment (DomT4) [inline Formula/tensorflow.rb]: We won't accept a new formula using `env :std`. Please explain why you need it, Thanks.

File: Formula/tensorflow.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    sha256 "1ebb8ad7e26b448e9caa4773d2357849bf80ff9e313964bcaf79cbf0201a1648"
+  end
+
+  resource "six" do
+    url "https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz"
+    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
+  end
+
+  def install
+    resource("wheel").stage { system "python", *Language::Python.setup_install_args(libexec/"vendor") }
```
Comment (DomT4) [inline Formula/tensorflow.rb]:
```ruby
    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
    resources.each do |r|
      r.stage do
        system "python", *Language::Python.setup_install_args(libexec/"vendor")
      end
    end
```

File: Formula/tensorflow.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    resource("six").stage { system "python", *Language::Python.setup_install_args(libexec/"vendor") }
+    ENV["PYTHON_BIN_PATH"] = which "python"
```
Comment (DomT4) [inline Formula/tensorflow.rb]: Is this not detected by the build process automatically?

File: Formula/tensorflow.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ENV["TF_NEED_CUDA"] = "0"
+
+    system "./configure"
+
+    system "bazel", "build", "-c", "opt", "//tensorflow/tools/pip_package:build_pip_package"
+    system "bazel-bin/tensorflow/tools/pip_package/build_pip_package", "/tmp/tensorflow_pkg"
```
Comment (DomT4) [inline Formula/tensorflow.rb]: Please keep temporary things installed within the `buildpath`.

File: Formula/tensorflow.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    system "pip install --ignore-installed --user /tmp/tensorflow_pkg/tensorflow-0.7.1-*.whl"
```
Comment (DomT4) [inline Formula/tensorflow.rb]: You can't use `pip` directly in a formula.

File: Formula/tensorflow.rb (hunk 1 above)
Comment (itsmeolivia) [inline Formula/tensorflow.rb]: I was getting an error that said "error: C compiler cannot create executables" and when looking through my error logs, I found something that suggested I set the env to std.  Is there a better way to do this?

File: Formula/tensorflow.rb (hunk 3 above)
Comment (itsmeolivia) [inline Formula/tensorflow.rb]: No, it's not. `./configure` requires user input for the `PYTHON_BIN_PATH` (see [these lines](https://github.com/tensorflow/tensorflow/blob/v0.7.1/configure#L4-L23)).

File: Formula/tensorflow.rb (hunk 1 above)
Comment (DomT4) [inline Formula/tensorflow.rb]: Can you provide build logs for that error? Thanks.

File: Formula/tensorflow.rb (hunk 5 above)
Comment (itsmeolivia) [inline Formula/tensorflow.rb]: still working on this fix!

File: Formula/tensorflow.rb (hunk 1 above)
Comment (itsmeolivia) [inline Formula/tensorflow.rb]: how do I do that? I tried brew install --verbose --debug --list-checks tensorflow > buildlog.txt 2>&1 and it just hung

Comment (apjanke) [inline Formula/tensorflow.rb]: After you abort the hung build with Ctrl-C, run `brew gist-logs tensorflow`. It'll collect the build logs that were generated behind the scenes and post them as a new gist on GitHub.

Comment (itsmeolivia) [inline Formula/tensorflow.rb]: https://gist.github.com/3d2910ac0b68e05e7bd966aabd3a6f5b

do you also want the config.log?

File: Formula/tensorflow.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    system "./configure"
+
+    system "bazel", "build", "-c", "opt", "//tensorflow/tools/pip_package:build_pip_package"
+    system "bazel-bin/tensorflow/tools/pip_package/build_pip_package", buildpath/"tensorflow_pkg"
+
+    system "pip install --ignore-installed --user #{buildpath}/tensorflow_pkg/tensorflow-0.7.1-*.whl"
+  end
+
+  def caveats; <<-EOS.undent
+    TensorFlow depends on the development version of Protobuf which cannot be installed automatically.
```
Comment (mikemcquaid) [inline Formula/tensorflow.rb]: We'll need to wait until this can work with a stable protobuf before we include it.

**Comment by apjanke** [comment]
Closing because we need to wait for it to work with a stable release of `protobuf`, and to not use `env :std`. Feel free to resubmit when those are resolved. In the meantime, you can add this formula to [your own tap](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md) to make it available via `brew`.

**Comment by nickdesaulniers** [comment]
@itsmeolivia , do you know if the dev version of protobuf is still required?

**Pull request closed without merging**